src/http/server.clj

```diff
@@ -293,6 +294,24 @@
          (log/infof "\"%s %s HTTP/1.1\" %s %s" method'  uri status len))
        response))))
 
+(defn pw-prompt []
+  (str (do (println "Password: ") (.readPassword (System/console)))))
+
+(defn maybe-inject-auth
```

Comment:
The same way we do for `wrap-if-modified`: if there's no need to change anything, we can simply return `handler` as is instead of runtime check on each request.

Reply:
Very good point - checking for auth repeatedly is wasteful - I will rework this.